src/pdf/base/XojCairoPdfExport.cpp

```diff
@@ -133,7 +133,7 @@ auto XojCairoPdfExport::createPdf(Path file, PageRangeVector& range) -> bool {
 
     int c = 0;
     for (PageRangeEntry* e: range) {
-        for (int i = e->getFirst(); i <= e->getLast(); i++) {
+        for (int i = e->getFirst() - 1; i <= e->getLast() - 1; i++) {
             if (i < 0 || i > static_cast<int>(doc->getPageCount())) {
```

Comment:
If you look a line below you can see pages outside range will be ignored so the range check is already in place effectively.

Reply:
Yes, but the range check is _incorrect_ and needs to be fixed. If I have `N` pages, then putting `N+1` into the custom range will trigger a crash. What happens is that the upper bound becomes `N` (zero-indexing), so the range check will think that this is acceptable since `doc->getPageCount() == N`. But there are only `N` pages, so the maximum index is `N-1`, not `N`.

Reply:
I get it, I thought you were talking about the for loop expression ;) misunderstanding